File: data-ingestion/src/db/models/runner_result.py
"""Runner result model."""
from sqlalchemy import Column, Integer, Float, ForeignKey
from sqlalchemy.orm import relationship
from src.db.base import Base


class RunnerResult(Base):
    """Individual runner's race result."""

    __tablename__ = "runner_results"
    __table_args__ = {'schema': 'racing'}

    id = Column(Integer, primary_key=True, autoincrement=True)
    runner_id = Column(Integer, ForeignKey('racing.runners.id'), nullable=False, unique=True, index=True)
    race_result_id = Column(Integer, ForeignKey('racing.race_results.id'), nullable=False, index=True)

    # Finish Position
    finish_position = Column(Integer)  # 1 = winner, 2 = second, etc.

    # Payoffs
    win_payoff = Column(Float)
    place_payoff = Column(Float)
    show_payoff = Column(Float)

    # Relationships
    runner = relationship("Runner", back_populates="result")
    race_result = relationship("RaceResult", back_populates="runner_results")

    def __repr__(self):
        return f"<RunnerResult(runner_id={self.runner_id}, position={self.finish_position})>"

File: backend-api/src/main/java/com/horseracing/api/entity/RunnerResult.java
package com.horseracing.api.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "runner_results", schema = "racing")
@Data
@NoArgsConstructor
public class RunnerResult {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "race_result_id", nullable = false)
    private RaceResult raceResult;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "runner_id", nullable = false)
    private Runner runner;

    @Column(name = "finish_position")
    private Integer finishPosition;

    @Column(name = "win_payoff")
    private Double winPayoff;

    @Column(name = "place_payoff")
    private Double placePayoff;

    @Column(name = "show_payoff")
    private Double showPayoff;
}